internal/ir/xds.go

```diff
@@ -792,3 +796,28 @@ type OpenTelemetryAccessLog struct {
 	Port       uint32            `json:"port"`
 	Resources  map[string]string `json:"resources"`
 }
+
+// JSONPatchConfig defines the configuration for patching a Envoy xDS Resource
+// using JSONPatch semantics
+// +k8s:deepcopy-gen=true
+type JSONPatchConfig struct {
+	// Type is the typed URL of the Envoy xDS Resource
+	Type string `json:"type"`
+	// Name is the name of the resource
+	Name string `json:"name"`
+	// Patch defines the JSON Patch Operation
+	Operation JSONPatchOperation `json:"operation"`
+}
+
+// JSONPatchOperation defines the JSON Patch Operation as defined in
+// https://datatracker.ietf.org/doc/html/rfc6902
+// +k8s:deepcopy-gen=true
+type JSONPatchOperation struct {
+	// Op is the type of operation to perform
+	Op string `json:"op"`
```

Comment:
use full name here?

Reply:
or rename to `Type`, and change to Enum?

Reply:
this nameing is tied to JSONPatch spec, since we marshal this and directly pass it to https://github.com/evanphx/json-patch#create-and-apply-a-json-patch

Reply:
`Operation.Op` that's odd.

Reply:
+1

Reply:
its part of the spec :)
https://datatracker.ietf.org/doc/html/rfc6902/#section-4

Reply:
I know it looks a bit awkward, but as Arko pointed out, it aligns with the RFC standard.

Below description is extracted from the RFC:

Operation objects MUST have exactly one "op" member, whose value indicates the operation to perform.  Its value MUST be one of "add", "remove", "replace", "move", "copy", or "test"; other values are  errors.  

The following examples are from the same RFC.
```
   { "op": "add", "path": "/a/b/c", "value": "foo" }
   { "path": "/a/b/c", "op": "add", "value": "foo" }
   { "value": "foo", "path": "/a/b/c", "op": "add" }
 ```
 
 So I think we can settle with the current implementation so we won't block vendor extensions replying on the EnvoyPatchPolicy. It's internal to EG anyway,  we can always go back and change it once we figure out a better way to handle it.